ggshield/iac/iac_scan_models.py

```diff
@@ -47,19 +51,31 @@ def mock_api_iac_diff_scan(
         scan_parameters: IaCScanParameters,
         extra_headers: Optional[Dict[str, str]] = None,
     ) -> Union[Detail, IaCDiffScanResult]:
-        scan = self.iac_directory_scan(
-            Path("."),
-            [],
-            scan_parameters,
-            extra_headers,
-        )
-        result = IaCDiffScanResult(
-            scan.id,
-            scan.type,
-            scan.iac_engine_version,
-            entities_with_incidents=IaCDiffScanEntities([], [], []),
-        )
-        result.status_code = 200
+        result: Union[Detail, IaCDiffScanResult]
+
+        try:
+            resp = self.post(
+                endpoint="iac_diff_scan",
+                extra_headers=extra_headers,
+                files={
+                    "reference": reference,
+                    "current": current,
+                },
+                data={
+                    "scan_parameters": IaCScanParametersSchema().dumps(scan_parameters),
+                },
+            )
+        except requests.exceptions.ReadTimeout:
+            result = Detail("The request timed out.")
+            result.status_code = 504
+        else:
+            # if is_ok(resp):
+            result = IaCDiffScanResultSchema().load(resp.json())
+            # else:
+            # result = load_detail(resp)
```

Comment:
Why are there comments here ?

Reply:
This is a pygitguardian mock it will be removed and moved to GGClient.